tls/s2n_async_pkey.c

```diff
@@ -409,3 +422,140 @@ S2N_RESULT s2n_async_pkey_sign_free(struct s2n_async_pkey_op *op)
 
     return S2N_RESULT_OK;
 }
+
+int s2n_async_pkey_op_get_op_type(struct s2n_async_pkey_op *op, s2n_async_pkey_op_type *type)
+{
+    POSIX_ENSURE_REF(op);
+    POSIX_ENSURE_REF(type);
+
+    switch (op->type) {
+        case S2N_ASYNC_DECRYPT:
+            *type = S2N_ASYNC_DECRYPT;
+            break;
+        case S2N_ASYNC_SIGN:
+            *type = S2N_ASYNC_SIGN;
+            break;
+        default:
+            POSIX_BAIL(S2N_ERR_ASYNC_OP_UNINITIALIZED);
```

Comment:
How do you know that the type is unknown because the op is uninitialized? Couldn't it be unknown for other reasons?

Reply:
I think the only way it could be invalid is if the OP is corrupted or un-initialized, since the OP is allocated and initialized internally to the start of an opreation